src/sshagent/ASN1Key.h

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2018 Toni Spets <toni.spets@iki.fi>
+ *  Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ASN1KEY_H
+#define ASN1KEY_H
+
+#include "OpenSSHKey.h"
+#include <QtCore>
+
+class ASN1Key
```

Comment:
I wonder if this should be a namespace. This class has no members. Private functions can be wrapped into an anonymous namespace.

Reply:
What benefits a namespace has over a class in this case? I understand it would *work* given the circumstances of the class.

Reply:
It's more expressive. A stateless class is not a class, therefore use a namespace.
namespaces also use ADL, but that's not really important here.

Reply:
Converted public facing functions to namespace and made all private members file static.

Reply:
There is no reason to make those functions static if they are free functions already.